test/test_foreach.py

```diff
@@ -817,9 +817,9 @@ def test_foreach_copy_with_multi_device_inputs(self, device, dtype, op):
     def test_autodiff(self, device, dtype, op, inplace):
         if not (op.supports_autograd or op.supports_forward_ad):
             self.skipTest("neither reverse mode nor forward mode supported")
-        if (not inplace) and op.has_no_out_of_place:
+        if (not inplace) and not op.supports_out:
             self.skipTest("out-of-place not implemented")
-        if inplace and op.has_no_in_place:
+        if inplace and not op.has_no_in_place:
```

Comment:
Is this change correct? The name of the arg `has_no_in_place` is just not descriptive at all... ugh

Reply:
Reverted.